internal/app/wwclient/root.go

```diff
@@ -64,33 +73,29 @@ func CobraRunE(cmd *cobra.Command, args []string) (err error) {
 		return
 	}
 	pid, err := pidfile.Write(PIDFile)
-	defer cleanUp()
-	if err != nil && pid == -1 {
-		wwlog.Warn("%v. starting new wwclient", err)
-	} else if err != nil && pid > 0 {
-		return errors.New("found pidfile " + PIDFile + " not starting")
+	if err != nil {
+		if pid > 0 { // wwclient is already running
+			return fmt.Errorf("%v: not starting", err)
+		} else { // the pidfile is stale
+			wwlog.Warn("%s: starting new wwclient", err)
+		}
 	}
+	defer cleanUp()
 
+	target := "/"
 	if os.Args[0] == path.Join(conf.Paths.WWClientdir, "wwclient") {
-		err := os.Chdir("/")
-		if err != nil {
-			return fmt.Errorf("failed to change dir: %w", err)
-		}
 		wwlog.Warn("updating live file system: cancel now if this is in error")
 		time.Sleep(5000 * time.Millisecond)
 	} else {
+		target = "/warewulf/wwclient-test"
```

Comment:
I wonder if this should be an option, not hard coded.  If for some reason wwclient is not put in /warewulf in production this will get triggered.

Reply:
If not, I think the warning message is sufficient to debug by someone attempting to do this.